src/AstTransforms.cpp

```diff
@@ -101,28 +101,36 @@ bool RemoveRelationCopiesTransformer::removeRelationCopies(AstTranslationUnit& t
             if (!isFact(*cl) && cl->getBodyLiterals().size() == 1u && bodyAtoms.size() == 1u) {
                 AstAtom* atom = bodyAtoms[0];
                 if (equal_targets(cl->getHead()->getArguments(), atom->getArguments())) {
-                    // we have a match but have to check that all arguments are either
-                    // variables or records containing variables
-                    bool onlyVars = true;
+                    // Requirements:
+                    // 1) (checked) It is a rule with exactly one body.
+                    // 3) (checked) The body consists of an atom.
+                    // 4) (checked) The atom's arguments must be identical to the rule's head.
+                    // 5) (pending) The rules's head must consist only of either:
+                    //  5a) Variables
+                    //  5b) Aggregates unpacked into variables
+                    // 6) (pending) Each variable must have a distinct name.
+                    bool onlyDistinctHeadVars = true;
+                    std::set<std::string> headVars;
+
                     auto args = cl->getHead()->getArguments();
-                    while (!args.empty()) {
+                    while (onlyDistinctHeadVars && !args.empty()) {
                         const auto cur = args.back();
                         args.pop_back();
-                        if (!isVar(*cur)) {
-                            if (isRec(*cur)) {
```

Comment:
Since `isRec` and `isVar` aren't used anymore, could you remove the lambda functions from the start of the method?